### DISCUSSION TITLE
Can a parse function material property depend on the old value of a variable?

### DISCUSSION THREAD

**Discussion by bylore** [idaholab/moose#24448] [Q&A General] [answered]
Hi, i want to create a material property depending on the last time step value of a variable in a **parse function** manner. Can you guys give some advice?

**Comment by GiudGiud** [comment, marked as answer]
Hello

You can use the coupledValueOld instead of the coupledValue to get the variable value at the last time step.

However, with ParsedMaterial, we have not implemented that. 
So you would have to either:
- modify the code
- use a trick. Using the SelfAux auxkernel you can copy a variable onto another variable. If you execute it on TIMESTEP_BEGIN, you will effectively "save" the previous step variable value

Guillaume

**Reply by bylore** [reply]
Thanks for your repley. I found that ParsedMaterial and DerivativeParsedMaterial has a parameter "material_property_names". If I define a material property calculated on some variable old value. And  define a ParsedMaterial depend on the material property.  Then I guess I can obtain a ParsedMaterial depend on the variable old value?

**Reply by GiudGiud** [reply]
you could do that
